db.go

```diff
@@ -58,31 +70,46 @@ type SessionDB interface {
 
 // Authenticator that simply stores identity/passwords in memory
 type dummyAuthenticator struct {
-	users     map[string]*dummyUser
+	users     map[UserId]*dummyUser
```

Comment:
This is not the appropriate design for a dummy authenticator. The smoking gun is a *dummyUser present in both dummyUserStore and dummyAuthenticator. Those two ought to be different. The authenticator should store only passwords, and the userstore should store the same thing that a DB-based userstore stores.